File: leetcode/c++/two-sum.cc

/* Leetcode problem 1: Two Sum. */

#include <unordered_map>
#include <vector>

using namespace std;

class Solution
{
public:
  vector<int>
  twoSum (vector<int> &nums, int target)
  {
    unordered_map<int, int> table;
    vector<int>::size_type size = nums.size ();

    for (vector<int>::size_type i = 0; i < size; ++i)
      {
        int need = target - nums[i];
        if (table.count (need))
          return { (int) i, table[need] };
        table[nums[i]] = i;
      }
    return {};
  }
};

File: leetcode/rust/two-sum.rs
/* Leetcode problem 1: Two Sum. */

use std::collections::HashMap;

struct Solution {}

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut table = HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            let need = target - num;
            if let Some(&need_index) = table.get(&need) {
                return vec![need_index as i32, i as i32];
            }
            table.insert(num, i);
        }
        vec![]
    }
}

fn main() {}
